### DISCUSSION TITLE
a few suggestions on how to improve the widgets

### DISCUSSION THREAD

**Discussion by johdcmlaker2000** [tetrane/reven-community#6] [Ideas]
Hi,

Is there a possibility to copy selected memory buffer as bytearray from memory window? It would be great to have that option, if not.

I noticed that it is very inconvenient to click on esp every time and there on every DWORD in the memory window. It would be very convenient to add a separate window to display stack frame, as it is done, for example, in x64dbg.

**Comment by dureuill** [comment]
Hi,

> Is there a possibility to copy selected memory buffer as bytearray from memory window?

Currently, you can copy the selected memory buffer in the hexdump view by using the Ctrl+C shortcut, e.g. in the attached screenshot I selected some memory and then copy pasting will put "ec9781ffffffffb0ee9781ffffffff0000000000" in your clipboard. Would that address the need? Otherwise, which format would you expect for the copied bytearray?

![hexdump](https://user-images.githubusercontent.com/41078892/145373099-196ebe25-8282-461c-9231-5ccf4d64fcf8.png)

> It would be very convenient to add a separate window to display stack frame, as it is done, for example, in x64dbg.

Thank you for the suggestion, we will consider it for future REVEN versions! How would you like the view to be? Would a hexdump tracking the current value of `esp` be a good first step, or do you expect the view dword-by-dword like in the attached x64dbg screenshot? Note that you can change the view of the hexdump so that its content is displayed by DWORD. 

![image](https://user-images.githubusercontent.com/41078892/145373239-78ad6103-9057-4871-b2f6-23aeec57b29e.png)

**Reply by johdcmlaker2000** [reply]
> Currently, you can copy the selected memory buffer in the hexdump view by using the Ctrl+C shortcut
dureuill, 

thanks, i guess i just couldn't do it because of this problem https://github.com/tetrane/reven-community/discussions/7

> How would you like the view to be? 

It would be good exactly as in the screenshot, especially the part at the top of the screenshot that allows you to quickly navigate through the arguments passed to the function, taking into account the conversion of calls and the difference between x86 and x64 processes.

**Reply by dureuill** [reply]
Thank you for the precise feedback, we'll consider such a widget in the future. Meanwhile, ...

> allows you to quickly navigate through the arguments passed to the function, taking into account the conversion of calls and the difference between x86 and x64 processes.

if you want to access the arguments of a function, you can use our newly released prototype widget:

Here's a [tweet to see it in action](https://twitter.com/lodurel/status/1466812893295517701). If you don't know the prototype of the target function, you can leave the prototype part blank, and it will fill everything as pointers. Currently you need to set the correct calling convention manually.

Does that widget help you seeing the arguments on the stack?
